tests/bluetooth/bsim_bt/bsim_test_eatt/src/main_reconfigure.c

```diff
@@ -0,0 +1,89 @@
+/* main_reconfigure.c - Application main entry point */
+
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common.h"
+
+#include <bluetooth/gatt.h>
+
+CREATE_FLAG(flag_reconfigured);
+
+void att_mtu_updated(struct bt_conn *conn, uint16_t tx, uint16_t rx)
+{
+	printk("MTU Updated: tx %d, rx %d\n", tx, rx);
+	SET_FLAG(flag_reconfigured);
+}
+
+static struct bt_gatt_cb cb = {
+	.att_mtu_updated = att_mtu_updated,
+};
+
+static void test_peripheral_main(void)
+{
+	peripheral_setup_and_connect();
+
+	while (bt_eatt_count(default_conn) < CONFIG_BT_EATT_MAX) {
+		k_sleep(K_MSEC(10));
+	}
+
+	bt_gatt_cb_register(&cb);
```

Comment:
I registered the callback after all EATT channels was connected because it may be called when a channel is connected if the MTU increases. Testing locally, it seems to work with registering the callback before the loop, so can move it.